implementation/src/main/java/com/teampc/model/admin/User.java

```diff
@@ -143,5 +143,27 @@ public static User fromString(String serialized) {
       }
       return user;
    }
+
+   public UserType getUserType() {
```

Comment:
If we're going to delegate returning what type a user is to the actual object, there's no need to use the visitor. just declare an abstract `getType` method on the user object and the student and teacher can override it to return there types. I originally removed this `getType` because clients generally don't need to know if a user is a teacher or student, they just need to to tell the user what to do if it is a student and what to do if it is a teacher (hence the visitor). I don't really care about "good practices" this late in the game though